pkg/resource/deploy/step_generator_test.go

```diff
@@ -528,3 +531,432 @@ func TestDeleteProtectedErrorUsesCorrectQuotesOnOS(t *testing.T) {
 		assert.Contains(t, gotErrMsg, contains)
 	})
 }
+
+func TestGenerateReadSteps_read(t *testing.T) {
```

Comment:
I would have thought a lot of these paths were covered by the tests in engine/lifecycletests already? And if not they probably _should_ be lifecycle tests rather than step_generator unit tests.